src/NuGet.Core/NuGet.Protocol/Plugins/IPluginFactory.cs

```diff
@@ -11,20 +11,20 @@ namespace NuGet.Protocol.Plugins
     /// <summary>
     /// A plugin factory.
     /// </summary>
-    public interface IPluginFactory : IDisposable
+    internal interface IPluginFactory : IDisposable
```

Comment:
Can you explain this change?

Reply:
We had to update the API for IPluginFactory so that it is able to create plugins for dotnet tools plugins. Since we needed to break the API, I used it as an opportunity to un-expose the interface. The reason for making it internal is that the interface only has one implementation and there is no need for it to be a public API https://github.com/NuGet/NuGet.Client/pull/6113#discussion_r1830480428